src/dynamite/subspaces.py

```diff
@@ -221,6 +221,11 @@ def state_to_idx(self, state):
     @classmethod
     def _idx_to_state(cls, idx, L):
         idx = cls._numeric_to_array(idx)
+        dim = Full._get_dimension(L)
+        invalid_idx = idx[np.where(np.logical_or(idx >= dim, idx < 0))[0]]
```

Comment:
maybe use `max` and `min` to check so that we don't need to reallocate a large array every time